### PR TITLE
Move packages within src and fix extra things

### LINKED ISSUE #290: `netcomp` not working properly
Assuming have copied [config](https://imperialcollegelondon.github.io/SWMManywhere/config_guide/#minimum-viable-configuration) and updated accordingly. @dalonsoa it'd be amazing if you could take a look please.

```bash
python -m venv ./sa_env
./sa_env/Scripts/activate.bat
pip install swmmanywhere
python -m swmmanywhere --config_path=minimum_viable_template.yml 

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\bdobson\Downloads\temp\sa_env\Lib\site-packages\swmmanywhere\__main__.py", line 8, in <module>
    from swmmanywhere import swmmanywhere
  File "C:\Users\bdobson\Downloads\temp\sa_env\Lib\site-packages\swmmanywhere\swmmanywhere.py", line 22, in <module>
    from swmmanywhere.metric_utilities import iterate_metrics, validate_metric_list
  File "C:\Users\bdobson\Downloads\temp\sa_env\Lib\site-packages\swmmanywhere\metric_utilities.py", line 21, in <module>
    import netcomp
ModuleNotFoundError: No module named 'netcomp'
```

### PR DESCRIPTION
**Pull request by dalonsoa**
# Description

Moves `swmmanywhere` and `netcomp` to a `src` folder. And then updates `pyproject.toml` and a variety of other paths here and there to work with this new layout.

Tests pass, documentation builds and the generated wheel contains the right things (both `swmmanywhere` and `netcomp` can be imported after installing the wheel on a fresh environment), but I have not check trying to run a full script so I guess there might be some hidden things that need fixing. 

Fixes #290 again

## Type of change

- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [x] All tests pass (eg. `python -m pytest`)
- [x] The documentation builds and looks OK (eg. `python -m sphinx -b html docs docs/build`)
- [x] Pre-commit hooks run successfully (eg. `pre-commit run --all-files`)

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

**Comment by barneydobson** [comment]
@dalonsoa I don't understand these `ruff` errors - if I correct them to what is in the failed workflow action - it moves them back when I run `pre-commit` on my machine...

**Comment by cheginit** [comment]
For `coverage` you need:

```toml
[tool.coverage.report]
exclude_lines = [
  "if TYPE_CHECKING:",
]
omit = [
  "**/__init__.py",
]
ignore_errors = true

[tool.coverage.paths]
source = [ "src", "*/site-packages" ]
omit = [
  "**/__init__.py",
]

[tool.coverage.run]
branch = true
source = [
  "swmmanywhere",
]
omit = [
  "**/__init__.py",
]
```

Also, instead of:

```toml
only-include = ["src"]
sources = ["src"]
```

you can just do:
```toml
packages = [ "src" ]
```

**Comment by dalonsoa** [comment]
> @dalonsoa I don't understand these `ruff` errors - if I correct them to what is in the failed workflow action - it moves them back when I run `pre-commit` on my machine...

Not sure. I've just re-run pre-commit locally and all works now.

**Comment by barneydobson** [comment]
I followed advice in : https://github.com/astral-sh/ruff/issues/5667 - not sure if that's what did it but the pre-commit seemed to correct itself now

**Comment by dalonsoa** [comment]
> For `coverage` you need:
> 
> ```toml
> [tool.coverage.report]
> exclude_lines = [
>   "if TYPE_CHECKING:",
> ]
> omit = [
>   "**/__init__.py",
> ]
> ignore_errors = true
> 
> [tool.coverage.paths]
> source = [ "src", "*/site-packages" ]
> omit = [
>   "**/__init__.py",
> ]
> 
> [tool.coverage.run]
> branch = true
> source = [
>   "swmmanywhere",
> ]
> omit = [
>   "**/__init__.py",
> ]
> ```

Could you elaborate why we need all of these?

**Comment by dalonsoa** [comment]
> I followed advice in : [astral-sh/ruff#5667](https://github.com/astral-sh/ruff/issues/5667) - not sure if that's what did it but the pre-commit seemed to correct itself now

I enabled precommit.ci, which heals some errors automatically, creating new commits with the fixes. See #296

**Comment by barneydobson** [comment]
> > I followed advice in : [astral-sh/ruff#5667](https://github.com/astral-sh/ruff/issues/5667) - not sure if that's what did it but the pre-commit seemed to correct itself now
> 
> I enabled precommit.ci, which heals some errors automatically, creating new commits with the fixes. See #296

Not sure that fixes this though - since it is failing locally but passing after pre commit heal?

**Comment by dalonsoa** [comment]
Runs locally for me with no problems on the latest version of this branch with no problems.

**Comment by dalonsoa** [comment]
Everything nice and green :)

image

**Comment by cheginit** [comment]
> Could you elaborate why we need all of these?

Coverage doesn't work nice with src-layout by default, so some of these options are required to make it correctly detect the lines and report. Also, I usually omit the init to clean up the report.

**Comment by dalonsoa** [comment]
Done!

**Comment by barneydobson** [comment]
I dunno - `pre-commit run --all-files` is still doing `ruff` corrections on this branch on my machine - I will try on a different machine, maybe I have some weirdness going on on here

**Comment by cheginit** [comment]
OK, there's an important missing block:

```toml
license-files = { paths = ["LICENSE", "src/netcomp/LICENSE.txt"] }
```

File: src/swmmanywhere/metric_utilities.py [hunk 1]
```diff
@@ -12,13 +12,13 @@
 import cytoolz.curried as tlz
 import geopandas as gpd
 import joblib
```
Comment (barneydobson) [inline src/swmmanywhere/metric_utilities.py:14]: this should be down with the `swmmanywhere` imports no? `ruff` is moving this in `pre-commit` on my machine back down to `swmmanywhere` but `precommit.ci` is moving it back on here... @cheginit do you have any ideas here as @dalonsoa is unavailable at the mo

Comment (barneydobson) [reply src/swmmanywhere/metric_utilities.py:14]: this should be for the `import netcomp` line

Comment (cheginit) [reply src/swmmanywhere/metric_utilities.py:14]: Is this about `isort`? If so, you need to specify what are the first-party packages.

```toml
lint.isort.known-first-party = [
  "swmmanywhere",
  "netcomp",
]
```

Comment (cheginit) [reply src/swmmanywhere/metric_utilities.py:14]: It shouldn't include `src`, just the name of the package. The pre-commit error sounds like an issue with the service.

Comment (barneydobson) [reply src/swmmanywhere/metric_utilities.py]: OK - shall I just abandon `precommit.ci` and return it to what we had previously for now?

Comment (barneydobson) [reply src/swmmanywhere/metric_utilities.py]: oh lol it has passed now - maybe was a one off

**Review by barneydobson — APPROVED** [review]

File: src/swmmanywhere/metric_utilities.py (hunk 1 above)
Comment (cheginit) [reply src/swmmanywhere/metric_utilities.py]: Yeah, it sometimes happens, so don't sweat it 😄

**Pull request merged**